### DISCUSSION TITLE
Does the order of test methods matter when using NotInParallel?

### DISCUSSION THREAD

**Discussion by suugbut** [thomhurst/TUnit#711] [General]
When using `NotInParallel`, does the test runner **always** execute test methods from top to bottom? I ask this because it is not explicitly mentioned in the documentation.

```cs
public class MyTests
{
    private static int _value;

    [Test, NotInParallel]
    public void MyTest1() { _value = 99; }

    [Test, NotInParallel]
    public async Task MyTest2() { await Assert.That(_value).IsEqualTo(99); }
}
```

**Comment by thomhurst** [comment]
It should do. But for greater control you can use the DependsOn attribute.

**Reply by suugbut** [reply]
Thank you.
